bazel/execution_log_compact_to_csv.py

```diff
@@ -0,0 +1,108 @@
+# This Python script converts bazel's compact execution log (--execution_log_compact_file)
+# to a CSV file that contains a row per output file for every bazel target matching the given
+# --whitelist_pat regular expression. Each row contains the columns: target, path, and hash.
+#
+# Example usage:
+#
+# To invoke this script first generate a compact execution log with bazel using for example:
+#
+#   $ bazel build //rs/replica:replica --execution_log_compact_file=compact_execlog.zst
+#
+# Then decompress the file:
+#
+#   $ zstd -d compact_execlog.zst
+#
+# Finally build and run this script to convert the decompressed file to a CSV:
+#
+#   $ bazel build //bazel:execution_log_compact_to_csv && \
+#     bazel-bin/bazel/execution_log_compact_to_csv \
+#       --input_execlog=compact_execlog \
+#       --whitelist_pat='^//'
+#
+#   //rs/http_endpoints/public:build_script_,bazel-out/k8-opt-exec-ST-d57f47055a04/bin/rs/http_endpoints/public/build_script_,4463251579c194bf83a24408a184da9a7f46d5777f753e33a77afa5c2287fd1d
+#   //rs/http_endpoints/public:build_script,bazel-out/k8-opt/bin/rs/http_endpoints/public/build_script.out_dir/dashboard.rs,f0ba2e4976ac312782190b6391ce945f57708b46126b741f6336b355968a8a84
+#   ...
+#   //rs/replica:replica_lib,bazel-out/k8-opt/bin/rs/replica/libic_replica-974340861.rlib,0d45ec60aee4472020f69d0a19ac2ad4519388216f5a21e19152539fa3d0c4ce
+#   //rs/replica:replica,bazel-out/k8-opt/bin/rs/replica/replica,0ecf150fbff09dc7b92ac976827fc46179561f2e8e83487840330ce247b61b5d
+
+import argparse
+import csv
+import re
+import sys
+
+import google.protobuf.internal
+import spawn_pb2
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Convert bazel's compact execution log to a CSV file printed to stdout."
+    )
+    parser.add_argument(
+        "--input_execlog",
+        required=True,
+        help="Path to the zstd _decompressed_ input file as generated with bazel's --execution_log_compact_file option.",
+    )
+    parser.add_argument(
+        "--whitelist_pat",
+        default=None,
+        help="Regex to match target labels to include in the CSV. If omitted, all targets are included.",
+    )
+    parser.add_argument(
+        "--verbose",
+        action='store_true',
+        help="Log every ExecLogEntry to stderr. This is useful for debugging.",
+    )
+    args = parser.parse_args()
+
+    id_to_entry = {}
```

Comment:
can you add a brief comment explaining what this is used for?

Reply:
I'll move my comment from below over here:
```
            # We intend to log the outputs of Spawns. Spawns refer to their output by ID
            # so we need to keep track of the ID to entry mapping. Important note:
            # if an entry (like a Spawn) refers to another entry (like a File or Directory)
            # it's guaranteed the latter will precede the former.
```